index.js

```diff
@@ -50,7 +50,9 @@ YamlValidatore.prototype.validateStructure = function validateStructure(doc, str
 
   parent = parent || '';
 
-  Object.keys(structure).forEach(function eachKey(key) {
+  Object.keys(structure).forEach(function eachKey(originKey) {
+    const optional = originKey.endsWith('?');
+    const key = originKey.replace('?', '');
```

Comment:
Perhaps this should only replace the last character in case when it is `?`.

`/\?$/`